powershell/Tests/L0/Set-TaskVariable.SetsValue.ps1

```diff
@@ -5,19 +5,20 @@ param()
 . $PSScriptRoot\..\lib\Initialize-Test.ps1
 Invoke-VstsTaskScript -ScriptBlock {
     $variableSets = @(
-        @{ Path = 'Env:Some_name' ; Name = 'Some.name' ; Expected = 'Some value' }
-        @{ Path = 'Env:agent.jobstatus' ; Name = 'agent.jobstatus' ; Expected = 'Some job status' }
+        @{ Path = 'Env:Some_name' ; Name = 'Some.name' ; Expected = 'Some value' ; IsSecret = $false }
+        @{ Path = 'Env:Some_name' ; Name = 'Some.name' ; Expected = 'Some value' ; IsSecret = $true }
+        @{ Path = 'Env:agent.jobstatus' ; Name = 'agent.jobstatus' ; Expected = 'Some job status' ; IsSecret = $false }
     )
     foreach ($variableSet in $variableSets) {
-        Unregister-Mock Write-Host
-        Register-Mock Write-Host
+        Unregister-Mock Write-SetVariable
+        Register-Mock Write-SetVariable
```

Comment:
I found out mocking Write-Host makes things difficult to read the test trace output. Switching over to mocking Write-SetVariable since it makes the assertion easier and doesn't mess up the trace output.

Also, I added IsSecret to the test cases. This was another reason to move away from asserting the value passed to Write-Host. The property bag within the logging command now contains two values (name and issecret). The problem with trying to assert the correct value was passed to write-host is that the property bag internally is passed to a private format method as a hashtable. Ordered hastable wasn't introduced until PowerShell 4, so the order that the properties are written is unpredictable.